### PR TITLE
Add LocalPath topic for website to visualize

### PR DESCRIPTION
**Pull request by jamenkaye**
### Description

Local Pathfinding would like to visualize paths on the website. This provides a ROS topic to communicate the path to the website.

### Verification

- The local path is just a list of waypoints, so the implementation is identical to the Global Path

Things I'd like to confirm:
- Do I need to do anything to indicate the topic is named `local_path` or does that just come up when the publisher is made? I couldn't find anything analogous with `global_path`
- Can someone please confirm the changes I made to the readme are accurate?

Also, should we rename the `GlobalPath` type to just `Path` or `WaypointList` or something, since the same type is used for Local and Global paths?

### CODE REVIEW

**Review by patrick-5546 — COMMENTED** [review]
> Do I need to do anything to indicate the topic is named local_path or does that just come up when the publisher is made?

You define it when creating the publisher

> Can someone please confirm the changes I made to the readme are accurate?

I think you forgot to update this field
image

> should we rename the GlobalPath type to just Path or WaypointList or something

Yea, I like `Path`.  You would also have to create a PR in local pathfinding to update https://github.com/UBCSailbot/local_pathfinding/blob/main/local_pathfinding/node_navigate.py

**Comment by jamenkaye** [comment]
Patrick pointed out that Pathfinding may want to communicate other data to the website to be visualized. This data can now be added to the LPathData type. As I'm still unsure exactly what form the other data will take, I haven't added it yet, so currently LPathData just includes a Path.

I'm not sure how to fix the failing test? I thought the colcon test was for tests in the `test` folder, but this repo doesn't seem to have one.

File: msg/LPathData.msg [hunk 1]
```diff
@@ -0,0 +1 @@
+Path localPath
```
Comment (patrick-5546) [inline msg/LPathData.msg]: Use snake case: `local_path`

**Comment by patrick-5546** [comment]
> I'm not sure how to fix the failing test?

It can't build, you can verify this by running the build package for custom_interfaces task locally.

You have to add your new msg here:
https://github.com/UBCSailbot/custom_interfaces/blob/d3c8af1e4c0fe6d7848031569187f81330e49785/CMakeLists.txt#L16-L35

File: README.md [hunk 2]
```diff
@@ -21,23 +21,24 @@ ROS messages and services used across many ROS packages in the project.
...
-| HelperHeading       | DesiredHeading, GPS, HelperAISShip |
-| HelperLatLon        | GlobalPath, GPS, HelperAISShip     |
-| HelperSpeed         | GPS, HelperAISShip, WindSensor     |
+| Interface           | Used In                                   |
+| ------------------- | ----------------------------------------- |
+| HelperAISShip       | AISShips                                  |
+| HelperBattery       | Batteries                                 |
+| HelperGenericSensor | GenericSensors                            |
+| HelperHeading       | DesiredHeading, GPS, HelperAISShip        |
+| HelperLatLon        | GlobalPath, GPS, LocalPath, HelperAISShip |
```
Comment (patrick-5546) [inline README.md]: - Rename GlobalPath to Path
- Remove LocalPath

**Review by patrick-5546 — APPROVED** [review]
Looks great, thanks Jamen!

After resolving the comment below feel free to merge. But local pathfinding CI will break once you do, so I would recommend doing so when you have time to create a PR to update local pathfinding with the GlobalPath->Path change.

@jahn18 FYI

File: README.md (hunk 2 above)
Comment (jamenkaye) [reply README.md]: Done! Sorry I missed these

File: msg/LPathData.msg (hunk 1 above)
Comment (jamenkaye) [reply msg/LPathData.msg]: Done!

**Pull request merged**